src/adapter/src/session.rs

```diff
@@ -109,6 +110,135 @@ where
     session_oracles: BTreeMap<Timeline, InMemoryTimestampOracle<T, NowFn<T>>>,
 }
 
+impl<T> SessionMetadata for Session<T>
+where
+    T: Debug + Clone + Send + Sync,
+    T: TimestampManipulation,
+{
+    fn user(&self) -> &User {
+        self.vars.user()
+    }
+
+    fn database(&self) -> &str {
+        self.vars.database()
+    }
+
+    fn search_path(&self) -> &[mz_sql_parser::ast::Ident] {
+        self.vars.search_path()
+    }
+
+    fn conn_id(&self) -> &ConnectionId {
+        &self.conn_id
+    }
+
+    fn pcx(&self) -> &PlanContext {
+        &self
+            .transaction()
+            .inner()
+            .expect("no active transaction")
+            .pcx
+    }
+
+    fn current_role_id(&self) -> &RoleId {
+        &self
+            .role_metadata
+            .as_ref()
+            .expect("role_metadata invariant violated")
+            .current_role
+    }
+
+    fn session_role_id(&self) -> &RoleId {
+        &self
+            .role_metadata
+            .as_ref()
+            .expect("role_metadata invariant violated")
+            .session_role
+    }
```

Comment:
Could you add a test case for this function? (Feel free to ignore if you consider this as not sensible.)